rdmo/management/rules.py

```diff
@@ -0,0 +1,155 @@
+import logging
+
+from django.db import models
+from django.contrib.sites.models import Site
+
+import rules
+
+logger = logging.getLogger(__name__)
+
+
+@rules.predicate
+def is_an_editor(user) -> bool:
+    ''' Checks if any editor role exists for the user '''
+    return user.role.editor.exists()
+
+
+@rules.predicate
+def is_multisite_editor(user) -> bool:
+    ''' Checks if the user is an editor for all the sites '''
+    if not is_an_editor(user):
+        logger.debug('rules.is_multisite_editor: user %s has no role editor', user)
+        return False
+    return user.role.editor.count() == Site.objects.count()
+
+
+@rules.predicate
+def is_element_editor(user, obj) -> bool:
+    ''' Checks if the user is an editor for the sites to which this element is editable '''
+    try:
+        if not obj.editors.exists():
+            # if the element has no editors, it is editable by all users with an editor role
+            return user.role.editor.exists()
+        return user.role.editor.filter(id__in=obj.editors.all()).exists()
+    except AttributeError:
+        logger.debug('rules.is_element_editor: obj %s has no attribute editors', obj)
+        return False
+
+
+@rules.predicate
+def is_a_reviewer(user) -> bool:
+    ''' Checks if any reviewer role exists for the user '''
+    return user.role.reviewer.exists()
+
+
+@rules.predicate
+def is_multisite_reviewer(user) -> bool:
+    ''' Checks if the user is an reviewer for all the sites '''
+    if not user.role.reviewer.exists():
```

Comment:
Is this only for the logs? I think this block is not really needed.

Reply:
for the implementation of this feature, we thought that `is_multisite_reviewer` or `is_multisite_editor` might be a special use-case for these roles. However, so far it doesn't look like this special case is required in the implementation yet.   

It could be used as a (quicker) "shortcut" for the permission checking but those optimizations can be done later as well..